doc/double-spend-proof-specification.md

```diff
@@ -0,0 +1,197 @@
+
+## Summary
+
+A double spend attack can be used, for instance, to redirect payments meant for a specific
+merchant to a different target and thus defraud the merchant we want to pay to. The basic
+concept of a double spend is that (at least) one unspent output is spent
+twice in different transactions which forces miners to pick one of them to mine.
+
+At its most basic we can detect this by finding two signed inputs both
+spending the same output. In the case of pay-to-public-key-hash (P2PKH)
+this means two signatures signing the same public key.
+
+Cryptographic signatures in Bitcoin Cash follow the 'fork-id' algorithm described
+[here](https://github.com/bitcoincashorg/bitcoincash.org/blob/master/spec/replay-protected-sighash.md),
+which explains a change made to the Satoshi designed algorithm, a change after which the containing transaction itself is not signed, but a unique hash of that
+transaction is being signed. This gives us the opportunity to send only
+the intermediate hashes instead of the whole transaction while allowing
+receivers to still validate both signatures of the same public
+key. And therefore prove that a double spend has taken place.
+
+## Limitation and risks
+
+Not all types and all combinations of transactions are supported. Wallets
+and point-of-sale applications are suggested to give a rating of how secure
+an unconfirmed transaction is based on various factors.
+
+Transactions that spend all, confirmed, P2PKH outputs with all inputs
+signed SIGHASH\_ALL without ANYONECANPAY, are double-spend-proof's
+"protected transactions".
+
+For more details see "*Using Double Spend Alerts: Merchant considerations*"
+below.
+
+## Motivation
+
+The chance of defrauding a merchant by double spending the transaction that is being
+paid to him is real and the main problem we have today is the fact that
+without significant infrastructure the merchant won't find out until the
+block is confirmed.
+
+The low risk of getting caught will make this a problem as Bitcoin Cash
+becomes more mainstream.
+
+The double-spend-proof is a means with which network participants that have
+the infrastructure to detect double spends can share that fact so merchants
+can receive information on their payment app (typically SPV based) in short enough
+time that the merchant can refuse to provide service or goods to their
+customer.
+
+
+## Network specification
+
+A node that finds itself in possession of a correct double-spend-proof
+shall notify its peers using the INV message, using a 'type' field with
+number **0x94a0**. This will be changed to another number as this spec
+is finalized.
+
+The hash-ID for the double-spend-proof is a double sha256 over the entire
+serialized content of the proof, as defined next.
+
+In response to an INV any peer can issue a `getdata` message which will
+cause a reply with the following message. The name of the message (until
+this spec is finalized) is **`dsproof-beta`**.
+
```

Comment:
@zander the code still has dsproof-beta...is it time to change this to just dsproof?